### ISSUE TITLE
Tests failing under debian8

### ISSUE THREAD

**Issue by c33s** [voxpupuli/puppet-php#433]
in multiple PR the tests fail with debian8

- https://github.com/voxpupuli/puppet-php/pull/432 https://travis-ci.org/voxpupuli/puppet-php/jobs/355132633
- https://github.com/voxpupuli/puppet-php/pull/431 https://travis-ci.org/voxpupuli/puppet-php/jobs/354785257
- https://github.com/voxpupuli/puppet-php/pull/292  https://travis-ci.org/voxpupuli/puppet-php/jobs/355585742

```
Error: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install php5-cli' returned 100: Reading package lists...
Building dependency tree...
Reading state information...
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
php5-cli : Depends: libdb5.1 but it is not installable
          Recommends: php5-readline but it is not going to be installed
E: Unable to correct problems, you have held broken packages.
Error: /Stage[main]/Php::Packages/Package[php5-cli]/ensure: change from purged to present failed: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install php5-cli' returned 100: Reading package lists...
Building dependency tree...
Reading state information...
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
php5-cli : Depends: libdb5.1 but it is not installable
          Recommends: php5-readline but it is not going to be installed
E: Unable to correct problems, you have held broken packages.
```

- in debian7 wheezy https://packages.debian.org/wheezy/php5-cli is build against https://packages.debian.org/wheezy/libdb5.1
- in debian8 jessie https://packages.debian.org/jessie/php5-cli against https://packages.debian.org/jessie/libdb5.3
- in debian9 stretch (which is currently not in the build map) there is no php5-cli until we integrate the sury repo with auto usage of it.

broken tests output:
```
Notice: /Stage[main]/Php::Repo::Debian/Apt::Key[php::repo::debian]/Apt_key[php::repo::debian]/ensure: created
Notice: /Stage[main]/Php::Repo::Debian/Apt::Source[source_php_wheezy-php56]/Apt::Setting[list-source_php_wheezy-php56]/File[/etc/apt/sources.list.d/source_php_wheezy-php56.list]/ensure: defined content as '{md5}5a480294344d27cef9c39f8a50be08ea'
Info: /Stage[main]/Php::Repo::Debian/Apt::Source[source_php_wheezy-php56]/Apt::Setting[list-source_php_wheezy-php56]/File[/etc/apt/sources.list.d/source_php_wheezy-php56.list]: Scheduling refresh of Class[Apt::Update]
Notice: /Stage[main]/Php::Repo::Debian/Apt::Source[dotdeb-wheezy]/Apt::Setting[list-dotdeb-wheezy]/File[/etc/apt/sources.list.d/dotdeb-wheezy.list]/ensure: defined content as '{md5}f117d407b06bee024f38f77dd7884d71'
Info: /Stage[main]/Php::Repo::Debian/Apt::Source[dotdeb-wheezy]/Apt::Setting[list-dotdeb-wheezy]/File[/etc/apt/sources.list.d/dotdeb-wheezy.list]: Scheduling refresh of Class[Apt::Update]
Info: Class[Apt::Update]: Scheduling refresh of Exec[apt_update]
Notice: /Stage[main]/Apt::Update/Exec[apt_update]: Triggered 'refresh' from 1 events
Notice: /Stage[main]/Php::Packages/Package[php5-cli]/ensure: created
```

broken sources output:
```
Notice: /Stage[main]/Php::Repo::Debian/Apt::Key[php::repo::debian]/Apt_key[php::repo::debian]/ensure: created
Notice: /Stage[main]/Php::Repo::Debian/Apt::Source[dotdeb-wheezy]/Package[apt-transport-https]/ensure: created
Notice: /Stage[main]/Php::Repo::Debian/Apt::Source[source_php_wheezy-php56]/Apt::Setting[list-source_php_wheezy-php56]/File[/etc/apt/sources.list.d/source_php_wheezy-php56.list]/ensure: defined content as '{md5}388f39972254556224b8aeb890c1e672'
Info: /Stage[main]/Php::Repo::Debian/Apt::Source[source_php_wheezy-php56]/Apt::Setting[list-source_php_wheezy-php56]/File[/etc/apt/sources.list.d/source_php_wheezy-php56.list]: Scheduling refresh of Class[Apt::Update]
Notice: /Stage[main]/Php::Repo::Debian/Apt::Source[dotdeb-wheezy]/Apt::Setting[list-dotdeb-wheezy]/File[/etc/apt/sources.list.d/dotdeb-wheezy.list]/ensure: defined content as '{md5}f097cbe4a1ea1fe281b93efd805303b3'
Info: /Stage[main]/Php::Repo::Debian/Apt::Source[dotdeb-wheezy]/Apt::Setting[list-dotdeb-wheezy]/File[/etc/apt/sources.list.d/dotdeb-wheezy.list]: Scheduling refresh of Class[Apt::Update]
Info: Class[Apt::Update]: Scheduling refresh of Exec[apt_update]
Notice: /Stage[main]/Apt::Update/Exec[apt_update]: Triggered 'refresh' from 1 events
Error: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install php5-cli' returned 100: Reading package lists...
```

it could be a problem of the order of the source lists and that `wheezy` release is forced.

**Comment by c33s** [comment]
it is unclear for me, why the order of the source lists change and why it works before on jessie.

if i use logic values for the repos and use the jessie dotdeb repo for jessie (and not wheezy) it works
https://travis-ci.org/voxpupuli/puppet-php/builds/355666460?utm_source=github_status&utm_medium=notification

what we need is 
- refactoring the repo management for debian (use sury for all supported debian versions and dotdeb or the debian repos)
- adding debian7, debian8 and debian9 to the build matrix with php 5.6, 7.0, 7.1
- add tests for more php versions 5.6, 7.0, 7.1
- rebase the open PR

then the tests should work

or the more risky way would be to merge the PR which only have the debian test broken and refactor the repo management later but before a new release

**Comment by c33s** [comment]
maybe we have to create a new major version for this because of a BC break

**Comment by bastelfreak** [comment]
We shouldn't aim for a breaking change in each release. Sometimes they are needed. That is totally fine if the related issue/PR has a proper title and explains whay/why something changes. This will be automatically parsed and added to the top of the CHANGELOG.md.

**Comment by c33s** [comment]
php version & repo matrix
| -      | debian 7   | debian 8           | debian 9           |
|--------|------------|--------------------|--------------------|
| name   | wheezy     | jessie             | stretch            |
| lts    | May 2018   | April/May 2020     | June 2022          |
| ships  | 5.4 (php5) | 5.6 (php5)         | 7.0 (php)          |
| dotdeb | 5.5, 5.6   | 5.6, 7.0           | -                  |
| sury   | -          | 5.6, 7.0, 7.1, 7.2 | 5.6, 7.0, 7.1, 7.2 |

**Comment by c33s** [comment]
i am thinking about dropping wheezy and integrated dotdeb support and set `sury` as default repo and only use the native php if the requested version matches.

**Comment by jkroepke** [comment]
https://github.com/voxpupuli/puppet-php/pull/430

**Comment by SimonHoenscheid** [comment]
Why is debian 8 referenced in acceptence/php_spec and acceptence/php56_spec. Because 5.6 is the default, it is already covered in acceptence/php_spec. acceptence/php56_spec is the breaking one. removing it will fix the builds.

**Comment by c33s** [comment]
@SimonHoenscheid we should still investigate the problem because the tests worked before. we should know which change made them break. as far as i know only PR with green tests where merged, so why are the test breaking now? also why the wrong lib `libdb5.1` required.
have a bad feeling about just removing the test without knowing whats the problem

**Comment by SimonHoenscheid** [comment]
The other question I am asking myself: what is the php56 test covering for debian 8 which is not covered by the php test?

**Comment by SimonHoenscheid** [comment]
It seams to me the docker container is the wrong one. The travis job identifies the container  as debian 8, but the dotdeb repo connected is a wheezy one. (debian 7) This can not work with the dependencies.

`Info: /Stage[main]/Apt/Apt::Setting[conf-update-stamp]/File[/etc/apt/apt.conf.d/15update-stamp]: Scheduling refresh of Class[Apt::Update]`

` 	 php5-cli : Depends: libdb5.1 but it is not installable`

**Comment by SimonHoenscheid** [comment]
The change which broke builds was part of commit c0cb2d072df4c9ebffc2ad0e891fb90e0ec28e7a on Febuary 17. I would guess it was just a mistake.

**Comment by SimonHoenscheid** [comment]
@c33s please close bug was fixed in #439
